lib/mayaUsd/ufe/UsdMaterialHandler.cpp

```diff
@@ -0,0 +1,52 @@
+// ===========================================================================
+// Copyright 2022 Autodesk, Inc. All rights reserved.
+//
+// Use of this software is subject to the terms of the Autodesk license
+// agreement provided at the time of installation or download, or which
+// otherwise accompanies this software in either electronic or hard copy form.
+// ===========================================================================
+
+#include "UsdMaterialHandler.h"
+
+#include <mayaUsd/ufe/UsdSceneItem.h>
+
+#include <pxr/usd/usdGeom/imageable.h>
+
+namespace MAYAUSD_NS_DEF {
+namespace ufe {
+
+UsdMaterialHandler::UsdMaterialHandler()
+    : Ufe::MaterialHandler()
+{
+}
+
+UsdMaterialHandler::~UsdMaterialHandler() { }
+
+/*static*/
+UsdMaterialHandler::Ptr UsdMaterialHandler::create()
+{
+    return std::make_shared<UsdMaterialHandler>();
+}
+
+//------------------------------------------------------------------------------
+// UsdMaterialHandler overrides
+//------------------------------------------------------------------------------
+
+Ufe::Material::Ptr UsdMaterialHandler::material(const Ufe::SceneItem::Ptr& item) const
+{
+    UsdSceneItem::Ptr usdItem = std::dynamic_pointer_cast<UsdSceneItem>(item);
+#if !defined(NDEBUG)
+    assert(usdItem);
```

Comment:
Better to use TF_VERIFY and return nullptr.  See
https://github.com/Autodesk/maya-usd/blob/42854b32c9c1cd706917e088b477119ca1d791ab/lib/mayaUsd/ufe/UsdShaderNodeDefHandler.cpp#L50